### ISSUE TITLE
Style/MethodCallWithArgsParentheses: EnforcedStyle: omit_parentheses doesn't work (even with examples from manual)

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#14494]
## Steps to reproduce the problem

```bash
$ cat > r
foo(bar)
baz qux

# bad
array.delete e

# good
array.delete(e)

# good
# Operators don't need parens
puts 1 if foo == bar

# good
# Setter methods don't need parens
foo.bar = baz

# okay with `puts` listed in `AllowedMethods`
puts 'test'

# okay with `^assert` listed in `AllowedPatterns`
assert_equal 'test', x
^D
$ cat > c
AllCops:
  NewCops: enable
  TargetRubyVersion: 3.4.5

Style/FrozenStringLiteralComment:
  Enabled: false

Style/MethodCallWithArgsParentheses:
  EnforcedStyle: omit_parentheses
  # EnforcedStyle: require_parentheses # Neither works
^D
$ rubocop -c c r
```

## Actual behavior

`1 file inspected, no offenses detected`

## Expected behavior

https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/MethodCallWithArgsParentheses

## RuboCop version

```
ruby      2025-08-01 30a20bc16
rubocop   2025-08-12 c34098ee0
System Version: macOS 15.6 (24G84)
Model Identifier: MacBookAir10,1
```

**Comment by koic** [comment]
The true intent of the problem is not clear. Can you provide the expected code and actual code?
https://github.com/rubocop/rubocop/blob/master/.github/ISSUE_TEMPLATE/bug_report.md

**Comment by koic** [comment]
In addition to the above code, set `EnforcedStyle: omit_parentheses` in .rubocop.yml.

```console
$ cat .rubocop.yml 
AllCops:
  NewCops: enable
  TargetRubyVersion: 3.4

Style/FrozenStringLiteralComment:
  Enabled: false

Style/MethodCallWithArgsParentheses:
  EnforcedStyle: omit_parentheses

$ bundle exec rubocop --only Style/MethodCallWithArgsParentheses -a
Inspecting 1 file
C

Offenses:

example.rb:1:4: C: [Corrected] Style/MethodCallWithArgsParentheses: Omit parentheses for method calls with arguments.
foo(bar)
   ^^^^^
example.rb:8:13: C: [Corrected] Style/MethodCallWithArgsParentheses: Omit parentheses for method calls with arguments.
array.delete(e)
            ^^^

1 file inspected, 2 offenses detected, 2 offenses corrected
```

This appears to be the expected behavior.

```diff
@@ -1,11 +1,11 @@
-foo(bar)
+foo bar
 baz qux
 
 # bad
 array.delete e
 
 # good
-array.delete(e)
+array.delete e
 
 # good
 # Operators don't need parens
```

**Comment by forthrin** [comment]
Ah! The cop has to be explicitly enabled. Assumed it was implicitly enabled when using other options, eg. `EnforcedStyle`.

Problem solved, but maybe there should be a warning for this. Or simply enable it if it has other options set?

```Warning: You have applied `EnforcedStyle`, but the cop is disabled by default.```

```yaml
Style/MethodCallWithArgsParentheses:
  Enabled: true # REQUIRED
  EnforcedStyle: omit_parentheses
```

**Comment by koic** [comment]
Hmm, it might be an interesting idea, but giving implicit meaning between parameters could introduce the risk of unintended behavior. Therefore, the current behavior regarding enabling cops will remain unchanged. Since the issue in the subject has been resolved for now, this will be closed.
